### ISSUE TITLE
Unable to extract Kirby's Adventure (NES) - TypeError: sequence item 0: expected str instance, int found

### ISSUE THREAD

**Issue by wyatt8740** [JanErikGunnar/vcromclaim#2]
NES: Kirby's Adventure (ID: 46413845)
    Extracted manual to NES/Kirby's Adventure/manual
    Traceback (most recent call last):
      File "/path/to/vcromclaim_py3/a/../wiimetadata.py", line 662, in <module>
        nand.scantickets()
      File "/path/to/vcromclaim_py3/a/../wiimetadata.py", line 571, in scantickets
        ext.extract()
      File "/path/to/vcromclaim_py3/a/../wiimetadata.py", line 62, in extract
        if self.extractrom(app, os.path.join(self.channeltype, self.name), app[-10:]): rom_extracted = True
      File "/path/to/vcromclaim_py3/a/../wiimetadata.py", line 110, in extractrom
        return funcs[self.channeltype](u8arc, gameOutputPath, self.name)
      File "/path/to/vcromclaim_py3/a/../wiimetadata.py", line 143, in extractrom_nes
        output = BytesIO(''.join(output))
    TypeError: sequence item 0: expected str instance, int found

Donkey Kong and Mario 2 (NES) both extracted correctly earlier in the run.

Might it be because of the single quote in "Kirby's" or something like that?

Note that this worked in the old Python 2 version.

**Comment by JanErikGunnar** [comment]
Thank you for your report!
I'll look into this issue later!
Meanwhile here are some workarounds:
- Remove the games save data, OR
- Remove line 139-144 from vcromclaim.py
Either *MAY* result in an exported ROM in which emulators wont let you save in, maybe depending on emulator.
The crashing code tries to update a flag in the ROM file header, to indicate that the ROM has battery-backed SRAM. vcromclaim only does this if there is save data present. I'm not really sure if emulatores actually check this flag or not.

**Comment by JanErikGunnar** [comment]
Issue is fixed! Kirby's Adventure (possibly other NES games with SRAM) should be properly exported now, with or without save data present.

As mentioned, NES ROM headers have a flag telling emulators whether the game used SRAM (battery save) or not.
I am not sure if emulators actually care about this flag or not - some may. If the flag is OFF, some emulators might not save your progress.

MOST games have this flag correctly set ON or OFF. On Kirby's Adventure, which has SRAM, it is incorrectly OFF.

vcromclaim originally had a hack that checked IF the game had some saved data, and the flag was OFF, the flag must be wrong, and is updated. This is the code that I broke when migrating to Python 3.

I now fixed that crash.

However the flag would still be incorrect if no save data is present. I fixed that as well, by adding Kirby's Adventure and some other games to a list of known games which has SRAM. The games on this list will always have the flag forced ON, regardless if there is any save data present or not. The code is still checking for save data as well, in case the game is not in the list.
